go-controller/pkg/ovn/master.go

```diff
@@ -908,5 +995,46 @@ func (oc *Controller) syncNodes(nodes []interface{}) {
 		if err := oc.deleteNode(nodeName, nodeSubnets.hostSubnet, nodeSubnets.joinSubnet); err != nil {
 			klog.Error(err)
 		}
+		//remove the node from the chassis map so we don't delete it twice
+		delete(chassisMap, nodeName)
 	}
+
+	for nodeName, chassisName := range chassisMap {
+		if chassisName != "" {
+			_, stderr, err = util.RunOVNSbctl("--if-exist", "chassis-del", chassisName)
+			if err != nil {
+				klog.Errorf("Failed to delete chassis with name %s for logical switch %s: stderr: %q, error: %v",
+					chassisName, nodeName, stderr, err)
+			}
+		}
+	}
+}
+
+func (oc *Controller) unmarshalChassisDataIntoMap(chData []byte) (map[string]string, error) {
+	//map of node name to chassis name
+	chassisMap := make(map[string]string)
+
+	type chassisList struct {
+		Data     [][]string
+		Headings []string
+	}
+	var mapUnmarshal chassisList
+
+	if len(chData) == 0 {
+		return chassisMap, nil
+	}
+
+	err := json.Unmarshal(chData, &mapUnmarshal)
+	if err != nil {
+		return chassisMap, err
+	}
```

Comment:
you should do `return nil, err` isn't it?

Reply:
The caller is expecting a valid chassisMap. So either we return an empty map or we add a check in the caller. I am ok with either. wdyt?

Reply:
All callers are checking for `err`. If err != nil, then they will not proceed. So, in the case of error we return `nil` for the data along with the actual `err`.